core/src/main/java/com/linecorp/armeria/common/metric/MoreMeters.java

```diff
@@ -38,8 +45,22 @@
  */
 public final class MoreMeters {
 
+    private static final Logger logger = LoggerFactory.getLogger(MoreMeters.class);
+
     private static final double[] PERCENTILES = { 0, 0.5, 0.75, 0.9, 0.95, 0.98, 0.99, 0.999, 1.0 };
 
+    private static final boolean MICROMETER_1_5;
+
+    static {
+        final Set<Method> methods = ReflectionUtils.getMethods(Builder.class, method ->
```

Comment:
Can you add a TODO with a link to the issue to clean this up?

Reply:
I still need this logic because we can set `serviceLevelObjectives` from 1.5. 🤔

Reply:
Ah guess it makes sense. Wonder if using new micrometer APIs are worth these fragile-looking branches vs just compiling against 1.3.x. For example, I think this sort of branch breaks graalvm.

Reply:
> I think this sort of branch breaks graalvm.

If it's so, then we should not branch but send a PR to fix this. 😄
Let me send a one for this.

Reply:
- `Build.class.getMethod(....);` will do the job I guess? No need to use Reflections library.
- I don't think this breaks GraalVM anymore since they have changed the default behavior of static initializer evaluation. Even if they evaluate this block at image build time, I think it's fine because Micrometer will be baked in together, no?

Reply:
Yeah it's probably OK nowadays. Just I don't see such branching on versions in static languages much 😅

Reply:
Changed to use `getMethod`.